### PR TITLE
Navigator: Land: Improve it for VTOL by taking breaking distance into account

### PR DESCRIPTION
**Pull request by Claudio-Chies**
### Solved Problem
When triggering a Land on a VTOL, the horizontal speed might not be zero after the back transition, resulting in a weird flying back movement, which comes from the fact that the setpoint is only updated until the transition is done, and not until it has come to a stop.

### Solution
I propose we adapt the target landing location location setpoint based on the current velocity, and possible acceleration and jerk in MC mode.

### Changelog Entry
```
Bugfix: VTOL Landing Location
Documentation: Landing location for VTOL is where it came to an full halt, and not where backtransition was completed.
```

### Test coverage
Tested in SITL & on [Hardware](https://review.px4.io/plot_app?log=629ef75e-c257-4a24-912b-527d2f9634e6)

### CODE REVIEW

**Comment by Jaeyoung-Lim** [comment]
I can confirm that this caused quite hairy situations when flying our VTOL, (quick loss of altitude after transitions)

File: src/modules/navigator/land.cpp [hunk 1]
```diff
@@ -86,10 +86,8 @@ Land::on_active()
 	    _navigator->get_vstatus()->in_transition_mode) {
 		struct position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
 
-		// create a virtual wp 1m in front of the vehicle to track during the backtransition
-		waypoint_from_heading_and_distance(_navigator->get_global_position()->lat, _navigator->get_global_position()->lon,
-						   _navigator->get_local_position()->heading, 1.f,
-						   &pos_sp_triplet->current.lat, &pos_sp_triplet->current.lon);
+		// create a wp in front of the VTOL until which the VTOL in multicopter mode is able to decelerate
```
Comment (sfuhrer) [inline src/modules/navigator/land.cpp:89]:
```suggestion
		// create a wp in front of the VTOL while in back-transition, based on MPC settings that will apply in MC phase afterwards
```

**Review by sfuhrer — COMMENTED** [review]
Seems like the right solution to me. Let's though first test if on real hardware before merging.

**Comment by Claudio-Chies** [comment]
Tested on Hardware, seems to work nicely
added [flight log](https://review.px4.io/plot_app?log=629ef75e-c257-4a24-912b-527d2f9634e6) to PR above

**Review by sfuhrer — APPROVED** [review]
Thanks!

**Pull request merged**